### PR TITLE
[WIP] Add a simple dropdown to choose milestone templates from

### LINKED ISSUE #370: Add templates for milestones to DApp
## Description of the Task:
Add a button to the UI that would allow users chose a optional template. It should be a dropdown that adds profiled text to the description input field depending on the template chosen. The file that should be modified is [`src/components/views/EditMilestone.jsx`](https://github.com/Giveth/giveth-dapp/blob/develop/src/components/views/EditMilestone.jsx).  The possible position for the dropdown are marked in the screenshot below (not sure which one would work the best):

screen shot 2018-05-29 at 16 46 10

The description should be checked in the UI (you can add custom validators to [`src/lib/validators.js`](https://github.com/Giveth/giveth-dapp/blob/develop/src/lib/validators.js)) before sent to feathers that it contains the appropriate template text.
### Templates that should be supported are:

**1. Reward DAO Template**
 - Intro: (optional)
_Intro text talking about what you’ve been working on and why._ 
 - Description:
_Description what has been achieved_
 - Proof:
_Bullet points with links_
 - Video:
_Embedded video coming from https://fame.giveth.io/_
 - Reward:
_Amount of points, their type and how much they are worth in ETH and when
Format should be a list of `<amount> <type> at <rate>/<type> which is <amount in ETH> ETH`_

**2. Regular Reward Template:**
 - Intro: (optional)
_Intro text talking about what you’ve been working on and why._ 
 - Description:
_Description what has been achieved_
_Optionally links to proof_
 - Video: (optional but required if description has no links)
_Embedded video coming from https://fame.giveth.io/_
_Intro text talking about what you’ve been working on and why._  
 - Amount:
_Amount in fiat_

**3. Expenses Template:**
- Description:
_Description why is this an expense_
- The milestone must have expense items and they must have date smaller than today's date

**4. Bounties Template:**
 - Intro: (optional)
_Intro text to about the bounty_ 
- What:
_What should be achieved_
 - Why:
_Why is this a bounty, what is the benefit._
 - Deadline: (optional)
_By when the bounty needs to be finished_
 - Link to Bounty: (optional)
_Link to the bounty on for example status open bounties, or issue on github..._

### Examples:
Some examples of milestones we already have (but do not necessarily comply with the templates)
**1. Reward DAO** 
- https://alpha.giveth.io/campaigns/Bub3WLo6jmlG8V6j/milestones/chi2fAKXX7w07ssV

**2. Regular Rewards**
 - https://alpha.giveth.io/campaigns/R1WkS0obautijkvJ/milestones/WafbNMNVSCPC1yzM
 - https://alpha.giveth.io/campaigns/R1WkS0obautijkvJ/milestones/Lzyi7LXmEMEhXeBD
 - https://alpha.giveth.io/campaigns/fzOahNwFVyY7qLTI/milestones/ei6ItvOYrBG2gS58

**3. Expenses**
- https://alpha.giveth.io/campaigns/NMhA6QLwfsUmPQld/milestones/RGmC3kVtED4frXA7
- https://alpha.giveth.io/campaigns/fzOahNwFVyY7qLTI/milestones/fj76cSuLaZt5fApb

**4. Bounties**
- https://alpha.giveth.io/campaigns/fzOahNwFVyY7qLTI/milestones/j8CsAITBGgBVZFwH

## Original list of requirements made by @YalorMewn:
https://drive.google.com/open?id=1fzw32FGbH5kdYSgLovf2pUhBXGyRy9LE7Ce9CWam-9I

### PR DESCRIPTION
**Pull request by anshumanv**
Fixes #370 

TODO:
* Add validation
* Finalise dropdown position

How this looks right now - 
image

### CODE REVIEW

**Comment by vojtechsimetka** [comment]
Deploy preview for *giveth-dapp* processing.

Built with commit 89960fc36b624ce5cfaedfbb939c8302ff79f094

https://app.netlify.com/sites/giveth-dapp/deploys/5b1c4d083672df37dc76ccb4

**Comment by vojtechsimetka** [comment]
Deploy preview for *giveth-dapp* processing.

Built with commit 89960fc36b624ce5cfaedfbb939c8302ff79f094

https://app.netlify.com/sites/giveth-dapp/deploys/5b1c4d080733d5044554142a

**Comment by vojtechsimetka** [comment]
Hey @anshumanv , thank you for working on this! I almost think it should be part of the description toolbar. Does it make sense to you? https://github.com/zenoamaro/react-quill#custom-toolbar

**Comment by anshumanv** [comment]
Totally makes sense, thanks for the reference @vojtechsimetka! :smile: 
Will make the changes soon.

**Comment by anshumanv** [comment]
Hey @vs77bb, thanks for the ping, I'll complete this today. :+1: :100:

**Comment by anshumanv** [comment]
Almost completed, I'll be removing the react-dropdown lib that we imported before, since quill already provides a default dropdown which looks pretty nice, what I'm stuck at is the dropdown handler for the templates, still finding a workaround over it, will update after this.

**Comment by anshumanv** [comment]
@vojtechsimetka removed the react-dropdown dependency and using a default quilljs dropdown on the toolbar, sorry for the delay, please check once. :smile: :+1:

**Comment by vojtechsimetka** [comment]
Awesome, thank you! One more question are you going to add the validators as well? Those should be some simple regexes running on the description.

**Comment by anshumanv** [comment]
@vojtechsimetka thanks! I gave a thought about adding validators, actually what's presently done is on selecting a template an HTML is set on the description box, I should simply grab the html tag that needs validation and run checks on it right? Or do you have any other way of going about it?

**Comment by vojtechsimetka** [comment]
Yeah all you need to do is check if the description contains the template keywords :).

**Comment by anshumanv** [comment]
@vojtechsimetka added simple validation checks for template keywords, please see. Also thanks for being so patient all along. :sweat_smile: 

image

**Comment by anshumanv** [comment]
@vs77bb It says just 14hrs left for this bounty on gitcoin, just so it doesn't cause any problems, I'm submitting it. :smile:

File: src/components/views/EditMilestone.jsx [hunk 1]
```diff
@@ -639,12 +643,49 @@ class EditMilestone extends Component {
...
     const form = this.form.current.formsyForm;
     // we only block routing if the form state is not submitted
     this.setState({ isBlocking: form && (!form.state.formSubmitted || form.state.isSubmitting) });
   }
 
+  validateMilestoneDesc(value) {
+    console.log(value, this.state.template);
+    if (this.state.template === 'Reward DAO') {
+      return (
+        value.includes('Intro') &&
```
Comment (vojtechsimetka) [inline src/components/views/EditMilestone.jsx]: This should be optional but I can fix it :)

**Comment by anshumanv** [comment]
Thanks. :sweat_smile: 

Looking forward to make more contributions @vojtechsimetka. :smile: 
cc @vs77bb :sparkles:

**Pull request merged**